Sources/ComposableArchitecture/Effect.swift

```diff
@@ -97,16 +97,14 @@ extension Effect {
   public static func future(
     _ attemptToFulfill: @escaping (@escaping (Result<Value, Error>) -> Void) -> Void
   ) -> Effect {
-    return deferred { () -> SignalProducer<Value, Error> in
-      return SignalProducer { observer, _ in
-        attemptToFulfill { result in
-          switch result {
-          case let .success(value):
-            observer.send(value: value)
-            observer.sendCompleted()
-          case let .failure(error):
-            observer.send(error: error)
-          }
+    SignalProducer { observer, _ in
```

Comment:
Unlike `Combine.Future`, `SignalProducer` is not eager, so no need to use `deferred` here.